lib/portlayer/attach/attach.go

```diff
@@ -70,6 +70,12 @@ func toggle(handle *exec.Handle, connected bool) (*exec.Handle, error) {
 	b := serial.GetVirtualDevice().Backing.(*types.VirtualSerialPortURIBackingInfo)
 
 	serviceURI := fmt.Sprintf("tcp://%s:%d", ip, constants.SerialOverLANPort)
+	proxyURI := fmt.Sprintf("telnet://%s:%d", ip, constants.SerialOverLANPort)
+
+	if b.ProxyURI == proxyURI && c.Connected == connected {
+		log.Debugf("Already in the desired state (connected: %t, proxyURI: %s)", connected, proxyURI)
+		return handle, nil
+	}
 
 	if b.ServiceURI == serviceURI && c.Connected == connected {
```

Comment:
Only one of proxyURI vs serviceURI is used if I recall correctly.
Are we keeping this because proxyURI isn't needed on ESX only? If that's the case are we suppressing launch of vspc for esx only?